conda-store-server/conda_store_server/_internal/environment.py

```diff
@@ -141,3 +144,40 @@ def _append_pip_packages(specification, packages):
         )
 
     return specification
+
+
+def filter_environments(
+    query: Query,
+    role_bindings: schema.RoleBindings,
+) -> Query:
+    """Filter a query containing environments and namespaces by a set of role bindings.
+
+    Parameters
+    ----------
+    query : Query
+        Query containing both environments and namespaces
+    role_bindings : schema.RoleBindings
```

Comment:
What are the allowed `RoleBindings`? Going by https://github.com/peytondmurray/conda-store/blob/fad647e86f9b9acd8ae4b9148f1cf8e1e970b5ff/conda-store-server/conda_store_server/_internal/schema.py#L33 It would appear that anything that matches `namespace` and `environment` bindings per https://github.com/peytondmurray/conda-store/blob/fad647e86f9b9acd8ae4b9148f1cf8e1e970b5ff/conda-store-server/conda_store_server/_internal/schema.py#L36 would be acceptable values?

Reply:
Allowed values for `RoleBindings` are dictionaries mapping strings to lists of strings, just like you'd find in the role bindings configuration inside `conda_store_config.py`. Here are some examples:

```python
{
    "foo*/bar*": ['viewer'],
}
```

Describes `viewer` level access for any environments starting with `bar` (the `*` is wildcard) in any namespace starting with `foo`.


```python
{
    "*/*": ['admin'],
}
```

Describes `admin` level access for any environments in any namespace.


```python
{
    "test/env": ['editor'],
    "*/python": ['viewer'],
}
```

Describes `editor` level access for the environment named `env` in the namespace named `test`. It also describes `viewer` level access for any environment named `python` in any namespace.